modules/core/test/test_mat.cpp

```diff
@@ -1754,3 +1754,24 @@ TEST(Mat_, template_based_ptr)
 }
 
 #endif
+
+namespace cvtest {
+
+typedef testing::TestWithParam<tuple<int, Size, int> > Mat_init;
+TEST_P(Mat_init, ones)
+{
+    int depth = get<0>(GetParam());
+    Size size = get<1>(GetParam());
+    int ch = get<2>(GetParam());
+
+    Mat m = Mat::ones(size, CV_MAKETYPE(depth, ch));
+    ASSERT_EQ(countNonZero(m != 1), 0);
+}
+
+INSTANTIATE_TEST_CASE_P(Core, Mat_init, Combine(
+        Values(CV_8U, CV_8S, CV_16S, CV_32S, CV_32F, CV_64F), // depth
+        Values(Size(3, 3), Size(16, 8)),  // input size
+        Values(1, 2, 3, 4)  // number of channels
```

Comment:
Does it work for matrices of more than 4 channels?

Reply:
@csukuangfj, it doesn't because of https://github.com/opencv/opencv/blob/7349b8f5cecfc6eb649b224e58da16d137c3c578/modules/core/src/matrix.cpp#L1168 .